src/routes/mod_creation.rs

```diff
@@ -498,6 +502,25 @@ async fn mod_create_inner(
             }
         }
 
+        let slug_modid_option: Option<ModId> =
+            serde_json::from_str(&*format!("\"{}\"", mod_create_data.mod_slug)).ok();
+        if let Some(slug_modid) = slug_modid_option {
+            let slug_modid: models::ids::ModId = slug_modid.into();
+            let results = sqlx::query!(
+                "
+                SELECT EXISTS(SELECT 1 FROM mods WHERE id=$1)
+                ",
+                slug_modid as models::ids::ModId
+            )
+            .fetch_one(&mut *transaction)
+            .await
+            .map_err(|e| CreateError::DatabaseError(e.into()))?;
+
+            if results.exists.unwrap_or(true) {
+                return Err(CreateError::SlugCollision);
+            }
+        }
+
```

Comment:
Make this higher up where all the other validation stuff is, that way files aren't uploaded and deleted, which is a waste of bandwidth and processing.